### PR TITLE
some jenkins image plugin curation

### PR DESCRIPTION
**Pull request by gabemontero**
@bparees ptal

### CODE REVIEW

File: using_images/other_images/jenkins.adoc [hunk 1]
```diff
@@ -417,3 +417,29 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
 See the https://github.com/openshift/jenkins-plugin/[the plug-in's README] for a detailed description of what is available.
+
+== OpenShift Client Plug-in
+
+The rapid evolution of both Jenkins (including Pipelines and the introduction of Global Tool
+Configuration in Jenkins V2) and OpenShift have had an effect on how to best provide OpenShift integration from
+Jenkins jobs.  The advantages of not requiring the `oc` binary to be included in the Jenkins path, a key design point
+for OpenShift Pipeline plug-in, have been largely mitigated.  And the lack of functionality with client
+side logic in addition to REST flows, such as seen with `oc new-app` or binary builds, is more of a detriment as usage
+of the OpenShift Pipeline plug-in has expanded.
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: i'm thinking we don't need so much justification in the docs.  "Here's the new thing, here's what it replaces, here are the advantages"

File: using_images/other_images/jenkins.adoc [hunk 2]
```diff
@@ -417,3 +417,29 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
+
+Wit these realizations in mind, the new experimental https://github.com/openshift/jenkins-client-plugin[OpenShift Client Plugin for Jenkins]
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: s/Wit/With/ but i think based on my comment above this whole sentence is likely to change.

**Review by bparees — CHANGES_REQUESTED** [review]

File: using_images/other_images/jenkins.adoc (hunk 1 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: rework pushed

File: using_images/other_images/jenkins.adoc (hunk 2 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: rework pushed

File: using_images/other_images/jenkins.adoc [hunk 3]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
 plug-in for creating CI/CD flows for {product-title} in Jenkins.
 
 See the https://github.com/openshift/jenkins-plugin/[the plug-in's README] for a detailed description of what is available.
+
+== OpenShift Client Plug-in
+
+The experiences gained working with users of the OpenShift Pipeline plug-in, coupled with the rapid evolution of both Jenkins
+and OpenShift, have provided valueable insight into how to best provide {product-title} integration from Jenkins jobs.
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: valuable

**Review by bparees — CHANGES_REQUESTED** [review]
one nit and lgtm.

File: using_images/other_images/jenkins.adoc (hunk 3 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: fixed - thanks

**Comment by gabemontero** [comment]
@openshift/team-documentation ptal

**Comment by gabemontero** [comment]
@openshift/team-documentation bump

3.6 is on the horizon now ...any ETA on when we can merge this?

File: using_images/other_images/jenkins.adoc [hunk 4]
```diff
@@ -396,10 +396,10 @@ documentation].
 
 For more details on the sample job included in this image, see this link:https://github.com/openshift/origin/blob/master/examples/jenkins/README.md[tutorial].
 
-== {product-title} Pipeline Plug-in
+== OpenShift Pipeline Plug-in
 
-The Jenkins image's list of pre-installed plug-ins includes a plug-in which
-assists in the creating of CI/CD workflows that run against an {product-title}
+The Jenkins image's list of pre-installed plug-ins includes the OpenShift Pipeline plug-in, which
+assists in the creating of CI/CD workflows in Jenkins that run against an {product-title}
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: s/creating/creation

File: using_images/other_images/jenkins.adoc [hunk 5]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
 plug-in for creating CI/CD flows for {product-title} in Jenkins.
 
 See the https://github.com/openshift/jenkins-plugin/[the plug-in's README] for a detailed description of what is available.
+
+== OpenShift Client Plug-in
+
+The experiences gained working with users of the OpenShift Pipeline plug-in, coupled with the rapid evolution of both Jenkins
+and OpenShift, have provided valuable insight into how to best provide {product-title} integration from Jenkins jobs.
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: s/best provide {product-title} integration from Jenkins jobs/integrate {product-title} from Jenkins jobs.

File: using_images/other_images/jenkins.adoc [hunk 6]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
+
+As such, the new experimental https://github.com/openshift/jenkins-client-plugin[OpenShift Client Plugin for Jenkins]
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: We need to include link:  before all external URL paths, so:
link:https://github.com/openshift/jenkins-client-plugin[OpenShift Client Plug-in for Jenkins]

(note the - in plug-in)

File: using_images/other_images/jenkins.adoc [hunk 7]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
+is now offered as a technical preview and is included in the OpenShift Jenkins images on Centos (docker.io/openshift/jenkins-1-centos7:latest
+and docker.io/openshift/jenkins-2-centos7:latest). The plugin is also available from the Jenkins Update Center.  Envisioned as the long term
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: *docker.io/openshift/jenkins-1-centos7:latest* and *docker.io/openshift/jenkins-2-centos7:latest*

Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: s/plugin/plug-in

Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: The OpenShift Client plug-in will eventually replace the OpenShift Pipeline plug-in as the tool for OpenShift integration from Jenkins jobs. The OpenShift Client Plug-in provides:

File: using_images/other_images/jenkins.adoc [hunk 8]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
+direction for OpenShift integration from Jenkins jobs that will ultimately replace OpenShift Pipeline plug-in, the OpenShift Client plug-in provides:
+
+- A Fluent sytled syntax for use in Jenkins Pipelines
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: Fluent-styled

File: using_images/other_images/jenkins.adoc [hunk 9]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
+- Use of and exposure to any option available with `oc`
+- Integration with Jenkins Credentials and Clusters
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: Jenkins credentials and clusters

File: using_images/other_images/jenkins.adoc [hunk 10]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
+- Continued support for classic Jenkins Freestyle jobs
+
+== OpenShift Sync Plug-in
+
+To facilitate {product-title} xref:../../dev_guide/builds/build_strategies.adoc#pipeline-strategy-options[Pipeline build strategy] for integration
+between Jenkins and {product-title}, the https://github.com/openshift/jenkins-sync-plugin[OpenShift Sync plug-in] monitors the
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: link:https://github.com/openshift/jenkins-sync-plugin[OpenShift Sync plug-in]

File: using_images/other_images/jenkins.adoc (hunk 5 above)
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: looks like a strange line break here

File: using_images/other_images/jenkins.adoc [hunk 11]
```diff
@@ -417,3 +417,25 @@ Jenkins job] that is pre-configured in the Jenkins image utilizes the
...
+API server of {product-title} for updates `*BuildConfigs*` and `*Builds*` that employ the Pipeline strategy and either creates
```
Comment (ahardin-rh) [inline using_images/other_images/jenkins.adoc]: Looks like a strange line break here.

    `BuildConfigs  and `Builds` 

(several instances)

**Comment by ahardin-rh** [comment]
@gabemontero Sorry for the delay. I made some comments. It looks like you also have to rebase before we can get this moving forward. Thanks!

File: using_images/other_images/jenkins.adoc (hunk 4 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

File: using_images/other_images/jenkins.adoc (hunk 6 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

File: using_images/other_images/jenkins.adoc (hunk 7 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

File: using_images/other_images/jenkins.adoc (hunk 8 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

File: using_images/other_images/jenkins.adoc (hunk 9 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

File: using_images/other_images/jenkins.adoc (hunk 10 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

File: using_images/other_images/jenkins.adoc (hunk 11 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: part of next push

File: using_images/other_images/jenkins.adoc (hunk 5 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: hmm ... not seeing with my editor ... I'll switch to a couple of others, see if it emerges

**Comment by gabemontero** [comment]
thanks @ahardin-rh !  ... updates and rebase pushed

**Comment by ahardin-rh** [comment]
@gabemontero Thanks!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../using_images/other_images/jenkins.adoc#using-images-other-images-jenkins[Other Images -> Jenkins]
|New sections about xref:../using_images/other_images/jenkins.adoc#client-plugin-in[{product-title} Client Plug-in] and xref:../using_images/other_images/jenkins.adoc#sync-plug-in[{product-title} Sync Plug-in].
%

**Pull request merged**